src/main/java/org/hl7/davinci/priorauth/Database.java

```diff
@@ -503,16 +510,17 @@ public boolean delete(String resourceType, String id, String patient) {
   /**
    * Delete all resources of a particular type.
    * 
-   * @param resourceType - the resource type to delete.
+   * @param table - the Table to delete from.
    * @return boolean - whether or not the resources were deleted.
    */
-  public boolean delete(String resourceType) {
-    logger.info("Database::delete(" + resourceType + ")");
+  public boolean delete(Table table) {
+    logger.info("Database::delete(" + table.value() + ")");
     boolean result = false;
-    if (resourceType != null) {
+    if (table != null) {
       try (Connection connection = getConnection()) {
-        PreparedStatement stmt = connection.prepareStatement("DELETE FROM " + resourceType + ";");
-        result = stmt.execute();
+        PreparedStatement stmt = connection.prepareStatement("DELETE FROM " + table.value() + ";");
+        stmt.execute();
+        result = stmt.getUpdateCount() > 0 ? true : false;
       } catch (SQLException e) {
         logger.log(Level.SEVERE, "Database::runQuery:SQLException", e);
       }
```

Comment:
This looks really nice.

Reply:
I realized it wasn't doing what we thought it was before. `stmt.execute()` returns `true` and `false` based on whether the return value is a `ResultSet` or `UpdateCount`. Now it checks the update actually occurred to return true or false